### PR TITLE
Added `deleteAllAt()` to remove list of indices

### PR DESCRIPTION
**Pull request by Alexandre2006**
As suggested in [this](https://stackoverflow.com/questions/76823646/deleting-multiple-items-by-index-in-flutter-hive) StackOverflow thread, I've added a function that allows multiple elements in a box to be deleted at the same time based off of a list of indices.

I've also written tests for the new function.

### CODE REVIEW

File: hive/lib/src/box/box_base_impl.dart
Comment (themisir) [inline hive/lib/src/box/box_base_impl.dart:178]: We already have `deleteAll` method. I would suggest resolving keys using the indices and then passing them to that method instead of calling deleteAt one by one.

**Review by themisir — CHANGES_REQUESTED** [review]

File: hive/lib/src/box/box_base_impl.dart
Comment (Alexandre2006) [reply hive/lib/src/box/box_base_impl.dart]: I've removed the sorting and any additional bulk.

Instead of calling `deleteAll`, I've used writeFrames, which is probably significantly more efficient than re-verifying that the key exists.

Comment (Alexandre2006) [reply hive/lib/src/box/box_base_impl.dart]: Not sure if you would consider this "resolved." Please let me know if you think that using `deleteAll()` is a better method than using `writeFrames()`, and I'll implement the fix right away.

**Comment by Alexandre2006** [comment]
Any updates on merging this PR?

**Comment by themisir** [comment]
Sorry for this one taking so long, I am not sure about merging the change as it doesn't add any new feature to the library. There's already existing methods for converting indices to keys and method for deleting items in bulk with keys. This is by no means a final decision, just my personal opinion. Maybe other maintainers may have different opinions.

**Comment by Alexandre2006** [comment]
Understood. I think this just improves developer experience in some cases.

**Pull request closed without merging**
